Framework/Geometry/test/BeamlineRayTracerTest.h

```diff
@@ -0,0 +1,615 @@
+};
+
+class BeamlineRayTracerTestPerformance : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static BeamlineRayTracerTestPerformance *createSuite() {
+    return new BeamlineRayTracerTestPerformance();
+  }
+  static void destroySuite(BeamlineRayTracerTestPerformance *suite) {
+    delete suite;
+  }
+
+  BeamlineRayTracerTestPerformance() {
+    // Instrument
+    m_inst = ComponentCreationHelper::createTestInstrumentRectangular(2, 100);
+
+    // Create an instrument visitor
+    InstrumentVisitor visitor{m_inst};
+
+    // Visit everything
+    visitor.walkInstrument();
+
+    // Get ComponentInfo and DetectorInfo objects and set them
+    auto infos = InstrumentVisitor::makeWrappers(*m_inst, nullptr);
+
+    // Unpack the pair
+    m_compInfo = std::move(infos.first);
+    m_detInfo = std::move(infos.second);
+
+    // Get the instrument
+    m_instTopaz = loadInstrumentDefinition("TOPAZ_Definition_2010.xml");
+
+    // Create an instrument visitor
+    InstrumentVisitor visitor2{m_instTopaz};
+
+    // Visit everything
+    visitor2.walkInstrument();
+
+    // Get ComponentInfo and DetectorInfo objects and set them
+    auto infos2 = InstrumentVisitor::makeWrappers(*m_instTopaz, nullptr);
+
+    // Unpack the pair
+    m_compInfoTopaz = std::move(infos2.first);
+    m_detInfoTopaz = std::move(infos2.second);
+
+    m_instWish = loadInstrumentDefinition("WISH_Definition_10Panels.xml");
+
+    // Create an instrument visitor
+    InstrumentVisitor visitor3{m_instWish};
+
+    // Visit everything
+    visitor3.walkInstrument();
+
+    // Get ComponentInfo and DetectorInfo objects and set them
+    auto infos3 = InstrumentVisitor::makeWrappers(*m_instWish, nullptr);
+
+    // Unpack the pair
+    m_compInfoWish = std::move(infos3.first);
+    m_detInfoWish = std::move(infos3.second);
+  }
+
+  std::vector<V3D> makeTestDirections() {
+    std::vector<V3D> testDirections;
+    testDirections.reserve((360 / 3) * (180 / 3));
+
+    // Directly in Z+ = towards the detector center
+    for (int azimuth = 0; azimuth < 360; azimuth += 3) {
+      for (int elev = -89; elev < 89; elev += 3) {
+        // Make a vector pointing in every direction
+        V3D testDir;
+        testDir.spherical(1, double(elev), double(azimuth));
+        testDirections.push_back(testDir);
+      }
+    }
+    return testDirections;
+  }
+
+  Instrument_sptr loadInstrumentDefinition(const std::string &idfName) {
+    // Parse test file
+    std::string filename =
+        ConfigService::Instance().getInstrumentDirectory() + idfName;
+    std::string xmlText = Mantid::Kernel::Strings::loadFile(filename);
+    InstrumentDefinitionParser IDP(filename, "UnitTesting", xmlText);
+
+    // Get the instrument
+    return IDP.parseXML(nullptr);
+  }
+
+  void test_RectangularDetector() {
+    std::vector<V3D> testDirections;
```

Comment:
Do you want to include this block in the timings? Might be misleading for future maintainers.